blockchain/types/block.go

```diff
@@ -409,6 +409,25 @@ func (b *Block) WithBody(transactions []*Transaction) *Block {
 	return block
 }
 
+// WithoutBlobTxSidecar returns a new block with the blob sidecar removed from the transactions.
+func (b *Block) WithoutBlobTxSidecar() *Block {
```

Comment:
I think `WithoutBlobTxSidecar` doesn't need return value.
```
// RemoveSidecarFromBlobTxs removes the blob sidecar from the blob transactions.
func (b *Block) RemoveSidecarFromBlobTxs() {
	transactions := make([]*Transaction, len(b.transactions))
	for i, tx := range b.transactions {
		transactions[i] = tx.WithoutBlobTxSidecar()
	}
	b.transactions = transactions
}
```

So, instead of looping tx list twice, we can simply put `block.RemoveSidecarFromBlobTxs`.
```
	if block.IsBlobTxIncluded() {
		block = block.WithoutBlobTxSidecar()
	}
```